config/initializers/permissions.rb

```diff
@@ -171,6 +171,24 @@
                      permissible_on: :project,
                      require: :member,
                      contract_actions: { projects: %i[copy] }
+
+      map.permission :edit_attribute_help_texts,
+                     {},
+                     permissible_on: :global,
+                     require: :loggedin,
+                     grant_to_admin: true
+
+      map.permission :view_grids,
+                     {},
+                     permissible_on: :global,
+                     require: :loggedin,
+                     grant_to_admin: true
+
+      map.permission :edit_grids,
+                     {},
+                     permissible_on: :global,
+                     require: :loggedin,
+                     grant_to_admin: true
```

Comment:
Previously we had a system in place, that not-existing permissions wer always marked as granted to admins. And this way the permission was silently granted to each admin:

https://github.com/opf/openproject/blob/fc6f39ff3d50b2a287730021781f11f5e37ffff9/lib/open_project/access_control.rb#L176-L180

As we are explicitly checking that all permissions exist now, this lead to errors. So, here we have added all permissions that were previously granted to admins behind the scenes.

Reply:
For the `edit_attribute_help_text` I can reproduce the problem and understand where it comes from. However I don't think that adding the permission is the right solution. It creates new problems by adding the permission to the list of global permissions:

<img width="430" alt="image" src="https://github.com/opf/openproject/assets/617519/d62eb210-6075-4385-b45c-3ada357e4caa">

But even with the permission checkbox checked, a non admin user cannot make use of the permission. This will in my opinion only cause confusion. 

I see four options:

1.  Extend the functionality so that non admins can actually manipulate `AttributeHelpText` models. They would need to have access to the controller and the contracts would have to be adjusted as well.
2. Introduce a specific option to `acts_as_journalized only_admin: true` so that only admins are allowed to manipulate thus avoiding the generic permission check.
3. Allow the `acts_as_journalized` options to take a proc that is evaluated rather then the permission check.
4. Misuse an existing global permission e.g. `:manage_user`. 

The last one is dirty without question. It has the potential of causing problems later on and is just surprising to do. The first one is the option that will yield the most benefits in the long run but might require some work. Option 2 and 3 are rather similar. Given the `viewable_by_all_users` option introduced apparently only for the `AttributeHelpText` functionality, if I had to choose between the two, I'd go for option 3 so the option can be removed again. Option 2 looks like it will complicate the code. 

My recommendation is to check if option 1 really boils down to only altering the permission checks in the controller and contract. If it does, I'd go with it. If not, I'd go for option 3.

Reply:
Looking into option 1.

Reply:
> Ideally, the superclass would complain if another subclass is added without specifying the permissions but for those that exists it is covered, so I don't think that we need to define those permissions. But I guess that you didn't simply add this by chance. When did you run into a `UnknownPermissionError` for that permission?

When removing the permissions in the grid.rb model, the following tests fail:

```
rspec './modules/grids/spec/models/grids/grid_spec.rb[1:1:2:3:1]' # Grids::Grid behaves like grid attributes behaves like acts_as_attachable included #attachments_visible allows access to a logged user when viewable_by_all_users is set
rspec './modules/grids/spec/models/grids/grid_spec.rb[1:1:2:1:4:1]' # Grids::Grid behaves like grid attributes behaves like acts_as_attachable included #validations on attachments_claimed for attachments that are attached, created by the current_user, added to attachments_claimed and the user having the permission is invalid
rspec './modules/grids/spec/models/grids/grid_spec.rb[1:1:2:1:3:1]' # Grids::Grid behaves like grid attributes behaves like acts_as_attachable included #validations on attachments_claimed for attachments that are unattached, created by the current_user, added to attachments_claimed and the user not having the permission is invalid
rspec './modules/grids/spec/models/grids/grid_spec.rb[1:1:2:1:2:1]' # Grids::Grid behaves like grid attributes behaves like acts_as_attachable included #validations on attachments_claimed for attachments that are unattached, created by the current_user, added to attachments_claimed and the user having the permission is valid
rspec './modules/grids/spec/models/grids/grid_spec.rb[1:1:2:1:5:1]' # Grids::Grid behaves like grid attributes behaves like acts_as_attachable included #validations on attachments_claimed for attachments that are unattached, not created by the current_user, added to attachments_claimed and the user having the permission is invalid
rspec './modules/grids/spec/models/grids/grid_spec.rb[1:1:2:2:1]' # Grids::Grid behaves like grid attributes behaves like acts_as_attachable included #attachments_claimed updates all attachments to be linked to the model before saving
```